core/server/src/main/java/alluxio/master/journal/ProtoBufJournalFormatter.java

```diff
@@ -40,15 +48,35 @@ public void serialize(JournalEntry entry, OutputStream outputStream) throws IOEx
   @Override
   public JournalInputStream deserialize(final InputStream inputStream) throws IOException {
     return new JournalInputStream() {
+      private final byte[] mBuffer = new byte[1024];
       private long mLatestSequenceNumber;
 
       @Override
       public JournalEntry getNextEntry() throws IOException {
-        JournalEntry entry = JournalEntry.parseDelimitedFrom(inputStream);
-        if (entry != null) {
-          mLatestSequenceNumber = entry.getSequenceNumber();
+        try {
+          int firstByte = inputStream.read();
+          if (firstByte == -1) {
+            return null;
+          }
+          int size = CodedInputStream.readRawVarint32(firstByte, inputStream);
+          byte[] buffer = size <= mBuffer.length ? mBuffer : new byte[size];
+          int bytes = inputStream.read(buffer, 0, size);
+          if (bytes < size) {
+            throw new InvalidProtocolBufferException(
+                "Journal entry was truncated. Expected to read " + size + " bytes but only got "
+                    + bytes);
+          }
```

Comment:
Should we only ignore in this case? What are the other possible causes of `InvalidProtocolBufferException`?